scipy/stats/_qmc.py

```diff
@@ -860,18 +968,56 @@ class LatinHypercube(QMCEngine):
            [6.80338249, 3.08795949],
            [2.65448791, 3.83491828]])
 
+    Alternatively, optimized LHS can be constructed with `optimization`.
+    It should give a sample with a lower discrepancy at a higher computational
+    cost:
+    >>> sampler = qmc.LatinHypercube(d=2, optimization="random-CD")
+    >>> sample = sampler.random(n=5)
+    >>> qmc.discrepancy(sample)
+    0.017579341838451112  # random
+
     """
 
     def __init__(
         self, d: IntNumber, *, centered: bool = False,
+        optimization: Optional[Literal["random-CD"]] = None,
         seed: SeedType = None
     ) -> None:
         super().__init__(d=d, seed=seed)
         self.centered = centered
 
+        optimization_methods: Dict[str, Callable] = {
+            "random-CD": self._random_optimal,
+        }
+
+        if optimization is None:
+            self.optimization = self._random
+        elif optimization in optimization_methods:
+            self.optimization = optimization_methods[optimization]
+        else:
+            raise ValueError(f"{optimization!r} is not a valid optimization"
+                             " method. It must be one of"
+                             f" {set(optimization_methods)!r}")
+
     def random(self, n: IntNumber = 1) -> np.ndarray:
         """Draw `n` in the half-open interval ``[0, 1)``.
 
+        Parameters
+        ----------
+        n : int, optional
+            Number of samples to generate in the parameter space. Default is 1.
+
+        Returns
+        -------
+        sample : array_like (n, d)
+            LHS sample.
+
+        """
+        return self.optimization(n)
```

Comment:
This is subjective, but I don't think it makes sense for `random` to always call `self.optimization`, which may just be `self._random`. Seems to me that `random` should produce a random sample (e.g. call `self._random`) and then, if `self.optimization` is not None, call the appropriate optimization method.

Reply:
Agreed. I wanted to avoid too much logic here. Alternatively I could replace the attribute's name with something like `lhs_method`. This way we still have a clear separation of both algorithms.

Reply:
You can keep the separation of you want. It could literally be just:
```
sample = self.random(n)
if self.optimization is not None:
    sample = self.optimization(sample)
```